llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -21874,6 +21874,108 @@ bool RISCVTargetLowering::fallBackToDAGISel(const Instruction &Inst) const {
   return false;
 }
 
+static MachineInstr *findInstrWhichNeedAllCSRs(MachineBasicBlock &MBB) {
+  // Some instructions may require (implicitly) all CSRs to be saved.
```

Comment:
Is it possible that there are other functions in addition to these functions that need all CSRs to be saved? Is it possible that there can be multiple MI in MBB that need this kind of handling?

Reply:
Doesn't callsUnwindInit need all CSR saved too?

Reply:
> Is it possible that there are other functions in addition to these functions that need all CSRs to be saved? Is it possible that there can be multiple MI in MBB that need this kind of handling?

It's possible, I only added the ones I know.

Reply:
Do we need to be worried about not handling cases that we don't currently know about? Do we need to solve this problem more generally?

Reply:
I don't know how we can solve this more generally. Maybe we can introduce a new attribute?